snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+using System;
+using System.Linq;
+using Xunit;
+using static Google.Datastore.V1Beta3.CommitRequest.Types;
+using static Google.Datastore.V1Beta3.Key.Types;
+using static Google.Datastore.V1Beta3.PropertyFilter.Types;
+using static Google.Datastore.V1Beta3.ReadOptions.Types;
+
+namespace Google.Datastore.V1Beta3.Snippets
+{
+    [Collection(nameof(DatastoreSnippetFixture))]
+    public class DatastoreClientSnippets
+    {
+        private readonly DatastoreSnippetFixture _fixture;
+
+        public DatastoreClientSnippets(DatastoreSnippetFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public void Lookup()
+        {
+            var projectId = _fixture.ProjectId;
+            var key1 = _fixture.PrideAndPrejudiceKey;
```

Comment:
We can ship at iterate. I just want to keep in mind why people are looking at this snippets.

On May 9, 2016, at 8:11 AM, Jon Skeet <notifications@github.com<mailto:notifications@github.com>> wrote:


In snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs<https://github.com/GoogleCloudPlatform/gcloud-dotnet/pull/139#discussion_r62515814>:

> +{
> +    [Collection(nameof(DatastoreSnippetFixture))]
> +    public class DatastoreClientSnippets
> +    {
> +        private readonly DatastoreSnippetFixture _fixture;
> +
> +        public DatastoreClientSnippets(DatastoreSnippetFixture fixture)
> +        {
> +            _fixture = fixture;
> +        }
> +
> +        [Fact]
> +        public void Lookup()
> +        {
> +            var projectId = _fixture.ProjectId;
> +            var key1 = _fixture.PrideAndPrejudiceKey;


True. What's slightly tricky is that the way we've got it here is "it's what was returned when we created it" - which probably isn't what users would be doing here.

I'm tackling key generation at the moment anyway, so maybe punt until that part is done (later this week)? If you'd really really like to see it in the first pass, maybe I need to expect PrideAndPrejudiceId instead, and show creating it from that and the partition id (which also needs to be shown, as per the other comment.)

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/GoogleCloudPlatform/gcloud-dotnet/pull/139/files/a0048ed36d028deafaea2fe2719af60e8a9fa09d#r62515814>